webapp/src/components/Metrics/PerformanceAnalysisTable.test.tsx

```diff
@@ -0,0 +1,230 @@
+import { fireEvent, screen, waitFor, within } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import PerformanceAnalysisTable from "./PerformanceAnalysisTable";
+import { AvailableDatasetSplits, DatasetSplitName } from "types/api";
+import { setupServer } from "msw/node";
+import {
+  getCustomMetricInfoAPIResponse,
+  getMetricsPerFilterAPIResponse,
+  getMetricsPerFilterAPIWithFailureResponse,
+} from "mocks/api/mockMetricsAPI";
+import {
+  getConfigAPIResponse,
+  getConfigMultipipelineAPIResponse,
+} from "mocks/api/mockConfigAPI";
+
+const setDatasetSplitName = jest.fn();
+const renderPerformanceAnalysisTable = (
+  availableDatasetSplits: AvailableDatasetSplits,
+  datasetSplitName: DatasetSplitName
+) =>
+  renderWithRouterAndRedux(
+    <PerformanceAnalysisTable
+      jobId="local"
+      pipeline={{ pipelineIndex: 0 }}
+      availableDatasetSplits={availableDatasetSplits}
+      isLoading={false}
+      datasetSplitName={datasetSplitName}
+      setDatasetSplitName={setDatasetSplitName}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("PerformanceAnalysisComparisonTableWithNoComparedPipeline", () => {
+  const handlers = [
+    getConfigAPIResponse,
+    getCustomMetricInfoAPIResponse,
+    getMetricsPerFilterAPIResponse,
+  ];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should display two toggle buttons for the train and test", () => {
+    renderPerformanceAnalysisTable({ train: true, eval: true }, "eval");
+    // verify if the toggles displayed with correct names
+    expect(
+      screen.getByRole("button", { name: "Evaluation Set", pressed: true })
+    ).toHaveValue("eval");
+    expect(
+      screen.getByRole("button", { name: "Training Set", pressed: false })
+    ).toHaveValue("train");
+  });
+
+  it("should not display pipeline select component if the config has not more than one pipeline", async () => {
+    renderPerformanceAnalysisTable({ train: true, eval: true }, "eval");
+    await waitFor(() =>
+      expect(screen.queryByText(/Compare Baseline/i)).toBeNull()
```

Comment:
This doesn't test much. It would pass instantly even if you searched for "Potato". I think we should at the very least `await` for something else that we expect to be there, then assert that "Compare Baseline" is not there.
```suggestion
      expect(screen.queryByText(/Potato/i)).toBeNull()
```

Reply:
Yeah, I could see it did not work as expected. I think I can ignore it as I have tested it [here](https://github.com/ServiceNow/azimuth/blob/babe44a4cc4068b1fd281619a1a2e0b218550b95/webapp/src/components/Metrics/PerformanceAnalysisTable.test.tsx#L135)